server/src/@common/dto/user/user.dto.ts

```diff
@@ -1,52 +1,94 @@
-import { User } from '@prisma/client';
-import { appConfig } from 'config/config';
+import { Profile, User } from '@prisma/client';
+import { appConfig } from '../../../../config/config';
 import { ERole, EBan } from '../../enums/user.enum';
+import { ApiProperty, PartialType } from '@nestjs/swagger';
+import { IsDate, IsEnum, IsInt, IsISO31661Alpha2, IsOptional, IsString } from 'class-validator';
+import { Exclude } from 'class-transformer';
+import { IsSteamCommunityID } from '../../validators/is-steam-id.validator';
+import { ProfileDto } from './profile.dto';
 
 export class UserDto implements User {
+    @ApiProperty()
+    @IsInt()
     id: number;
+
+    @ApiProperty()
+    @IsSteamCommunityID()
     steamID: string;
+
+    @ApiProperty()
+    @IsString()
     alias: string;
-    aliasLocked: boolean;
-    private _avatar: string;
+
+    @ApiProperty()
+    @IsEnum(ERole)
     roles: ERole;
+
+    @ApiProperty()
+    @IsEnum(EBan)
     bans: EBan;
+
+    @ApiProperty()
+    @IsISO31661Alpha2()
     country: string;
+
+    @ApiProperty()
+    @IsDate()
     createdAt: Date;
+
+    @ApiProperty()
+    @IsDate()
     updatedAt: Date;
 
+    private _avatar: string;
+
+    @ApiProperty()
     get avatarURL(): string {
-        const bans = this.bans;
-        const isAvatarBanned = bans & (1 << 2); // TODO: Refactor however needed to use Ban 'enum' (cyclic dep issue occurs when requiring user model in this file)
-        if (isAvatarBanned) {
+        // TODO: apparently steam's hosting has changed and this stuff is on cloudfare now, ask goc
```

Comment:
It has. The base URL is now:
`https://avatars.cloudflare.steamstatic.com/b0d98c45eb07061557a0ca85a7c3fb7cd24fda71_full.jpg`
where the old was:
`https://steamcdn-a.akamaihd.net/steamcommunity/public/images/avatars/b0/b0d98c45eb07061557a0ca85a7c3fb7cd24fda71_full.jpg`

So we should only be storing the `b0d98c45eb07061557a0ca85a7c3fb7cd24fda71_full.jpg` part now.

Reply:
Oh sorry, yeah I actually fixed this yesterday just hadn't include in this branch. Made some changes to the DTOs yesterday that I think are better but I wanted to work with them some more before committing, the URL change is in there.

Reply:
Okay I've pushed the changed DTOs I was doing including the URL change